calicoctl/commands/node/run.go

```diff
@@ -205,6 +203,21 @@ Description:
 		// the AS number, so convert.
 		asNumber = argutils.ValidateASNumber(asNumber).String()
 	}
+	// CLI flags will take precedence over env vars
```

Comment:
I think the auto-detection methods should also be there now that I think about it.